Web/Edubase.Services/Nomenclature/NomenclatureService.cs

```diff
@@ -43,6 +43,7 @@ public static string GetEstablishmentsPluralName(GT groupType, eTextCase textCas
                 case GT.MultiacademyTrust:
                 case GT.SingleacademyTrust:
                 case GT.SchoolSponsor:
+                case GT.SecureSingleAcademyTrust:
```

Comment:
I note that the address logic on line 325 has been updated to include Secure SAT, but a very similar line on at 363 has not.

- Does line 363 also need to be updated to include Secure SAT?
- Perhaps this repeated conversion of address warrants a short static utility method (or similar)?

https://github.com/DFE-Digital/get-information-about-schools/blob/secure_16-19_groups/Web/Edubase.Web.UI/Areas/Groups/Controllers/GroupController.cs#L325

```csharp
            var viewModel = new GroupDetailViewModel(_externalLookupService)
            {
                SearchQueryString = searchQueryString,
                SearchSource = searchSource,
                CanUserEdit = await _groupReadService.CanEditAsync(id, User),
                CanUserEditGovernance = await _groupReadService.CanEditGovernanceAsync(id, User),
                Group = model,
                GroupTypeName = model.GroupTypeId.HasValue ? await _lookup.GetNameAsync(() => model.GroupTypeId) : null,
                LocalAuthorityName = model.LocalAuthorityId.HasValue ? await _lookup.GetNameAsync(() => model.LocalAuthorityId) : null,
                GroupStatusName = model.StatusId.HasValue ? await _lookup.GetNameAsync(() => model.StatusId, "Group") : null,
/* HERE -> */   Address = model.GroupTypeId.OneOfThese(GT.SingleacademyTrust, GT.MultiacademyTrust, GT.ChildrensCentresGroup, GT.SecureSingleAcademyTrust) ? model.Address.ToString() : null,
                IsUserLoggedOn = User.Identity.IsAuthenticated,
                GroupTypeId = model.GroupTypeId ?? -1,
                IsClosed = model.StatusId == (int)eLookupGroupStatus.Closed || model.StatusId == (int)eLookupGroupStatus.CreatedInError,
                IsClosedInError = model.StatusId == (int)eLookupGroupStatus.CreatedInError,
                CloseDate = model.ClosedDate,
                UKPRN = model.UKPRN.ToInteger(),
            };
```

https://github.com/DFE-Digital/get-information-about-schools/blob/secure_16-19_groups/Web/Edubase.Web.UI/Areas/Groups/Controllers/GroupController.cs#L363

```csharp
            var viewModel = new GroupDetailViewModel(_externalLookupService)
            {
                Group = model,
                GroupTypeName = model.GroupTypeId.HasValue ? await _lookup.GetNameAsync(() => model.GroupTypeId) : null,
                LocalAuthorityName = await _lookup.GetNameAsync(() => model.LocalAuthorityId),
                GroupStatusName =  await _lookup.GetNameAsync(() => model.StatusId, "Group"),
/* HERE -> */   Address = model.GroupTypeId.OneOfThese(GT.SingleacademyTrust, GT.MultiacademyTrust, GT.ChildrensCentresGroup) ? model.Address.ToString() : null,
                IsUserLoggedOn = User.Identity.IsAuthenticated,
                GroupTypeId = model.GroupTypeId ?? -1,
                IsClosed = model.StatusId == (int) GS.Closed || model.StatusId == (int) GS.CreatedInError,
                IsClosedInError = model.StatusId == (int) GS.CreatedInError,
                CloseDate = model.ClosedDate,
                ChangeHistory = await _groupReadService.GetGovernanceChangeHistoryAsync(id, skip, 100, sortBy, User)
            };
```

Reply:
Good spot Roger you should be a tester!  Updated the could but will probably push the changes after I do all of it.
Not sure of the purpose of this tbh.  It does not look good.